packages/Button/src/button.scss

```
@@ -100,45 +103,67 @@
   padding: 0 var(--ids-btn-padding);
 }
 
-.ids-btn::after {
-  content: '';
-  position: absolute;
-  display: block;
-  pointer-events: none;
-  border-radius: var(--ids-btn-border-radius);
-  top: var(--ids-btn-focus-position);
-  left: var(--ids-btn-focus-position);
-  right: var(--ids-btn-focus-position);
-  bottom: var(--ids-btn-focus-position);
-  z-index: var(--ids-btn-z-index);
-  transition: box-shadow 0.1s linear;
+.ids-btn > span {
+  min-width: 0;
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
 }
 
+.ids-btn:focus,
+.ids-btn.focus,
 .ids-btn:hover {
   cursor: pointer;
   background: var(--ids-btn-background-primary-hover);
 }
 
+.ids-btn:disabled {
+  pointer-events: none;
+  user-select: none;
+  background: var(--ids-btn-background-primary-disabled);
+}
+
 .ids-btn:focus,
-.ids-btn:focus::after {
+.ids-btn.focus {
   outline: 0;
+
+  &:not(.ids-btn--loading) {
+    box-shadow: var(--focus);
+  }
+}
+
```

Afin de retirer l'effet de focus lors du click sur le bouton ajout ce bout de code 
```suggestion
.ids-btn:focus:not(:focus-visible) {
  box-shadow: none;
}
```